api/scripts/arborescence-monitoring/time-series.js

```diff
@@ -0,0 +1,64 @@
+import { assertNotNullOrUndefined } from '../../src/shared/domain/models/asserts.js';
+
+const MAXIMUM_NUMBER_OF_POINTS_THRESHOLD = 28;
+
+const PURGE_POINT_INDEX = 10;
+
+const NB_POINTS_TO_PURGE = 1;
+
+const NULL_OR_UNDEFINED_POINTS_ERROR_MESSAGE = 'Time series should have a non null/undefined array of points';
+
+export class TimeSeries {
+  #points;
+  constructor(points) {
+    assertNotNullOrUndefined(points, NULL_OR_UNDEFINED_POINTS_ERROR_MESSAGE);
+    points.sort((pointA, pointB) => new Date(pointA.x).getTime() - new Date(pointB.x).getTime());
+    this.#points = points;
+  }
+
+  size() {
+    return this.#points.length;
+  }
+
+  get(pointIndex) {
+    return this.#points[pointIndex];
+  }
+
+  toString() {
+    return JSON.stringify(this.#points);
+  }
+
+  add({ x, y }) {
+    if (this.#sameValueThanTheLast(y)) {
+      return this.#replaceLastPointByNewPoint({ x, y });
+    } else {
+      this.#points.push({ x, y });
+      if (this.#exceedMaxNumberOfPoints()) {
+        this.#purgeOnePoint();
+      }
+      return new TimeSeries(this.#points);
+    }
+  }
+
+  #purgeOnePoint() {
+    this.#points.splice(PURGE_POINT_INDEX, NB_POINTS_TO_PURGE);
+  }
```

Comment:
La fonction ne correspond pas à ce qu'elle fait

soit renommer

```suggestion
  #purgePoints() {
    this.#points.splice(PURGE_POINT_INDEX, NB_POINTS_TO_PURGE);
  }
```

soit ajuster son comportement

```suggestion
  #purgePoints() {
    this.#points.shift();
  }
```

Reply:
C'est bien purgeOnePoint, avec la constante à 1 ça correspond, ça retire un point à une position précise. Peut-être que j'interprète mal le splice...

Reply:
Pour essayer de formuler autrement

* Soit c'est `purgeOnePoint` et dans ce cas la variable  `NB_POINTS_TO_PURGE` ne sert à rien, c'est à dire que `this.#points.splice(PURGE_POINT_INDEX, 1);` c'est très bien ainsi et la fonction fait exactement ce qu'elle dit qu'elle va faire

* Soit c'est "purge moi un nombre de point variable `NB_POINTS_TO_PURGE`" et dans ce cas je propose `purgePoints`, qui fait que si demain NB_POINTS_TO_PURGE=2, la fonction aura un bon nom.

Reply:
Il ne s'agit pas de variable mais des constantes que j'ai extraites pour que ce soit plus explicite (en gros ce que signifie le 1 afin d'éviter un magic number;)).

Reply:
Renommer les constantes en PURGE_MIDDLE_POINT_INDEX, ONE_POINT_TO_PURGE